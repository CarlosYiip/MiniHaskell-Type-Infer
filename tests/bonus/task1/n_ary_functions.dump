[Bind "main" Nothing [] 

(Let [Bind "sum" Nothing [] (Letfun (Bind "sum" Nothing ["x","y"] 

(App (App (Prim Add) (Var "x")) (Var "y"))


))] (App (App (Var "sum") (Num 4)) (Num 4)))


]
