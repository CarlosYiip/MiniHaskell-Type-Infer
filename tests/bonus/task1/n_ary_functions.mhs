main
	= let sum
		= letfun sum x y z = x + y + z;
	in sum 4 4 4;
