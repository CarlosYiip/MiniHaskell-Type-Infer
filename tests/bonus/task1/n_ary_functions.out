main :: Int = let sum :: (Int -> Int -> Int -> Int) = (letfun sum :: (Int -> Int -> Int -> Int) x y z = (+) ((+) x y) z);
              in sum 4 4 4;